File: tests/fixtures/textmate/swift/basic.swift
import Foundation

// Swift basic fixture: café, 東京, λ, 🚀, and astral 𝌆.
enum Stage: String {
    case queued, ready, complete
}

struct Launch<Payload: CustomStringConvertible> {
    let name: String
    let stage: Stage
    let payload: Payload

    func summary(prefix: String) -> String {
        return "\(prefix) \(name): \(payload) 🚀 𝌆"
    }
}

func describe(_ stage: Stage) -> String {
    switch stage {
    case .queued: return "waiting"
    case .ready: return "ready"
    case .complete: return "done"
    }
}

print(Launch(name: "café 東京 λ", stage: .ready, payload: 42).summary(prefix: "Mission"))

File: tests/fixtures/textmate/kotlin/basic.kt
package fixtures.kotlin.basic

/* Basic Kotlin fixture:
 * café λ 🚀 𝌆
 */
data class Pilot(val name: String, val active: Boolean = true)

fun describe(pilot: Pilot?): String {
    val status = when {
        pilot == null -> "missing"
        pilot.active -> "ready"
        else -> "resting"
    }
    return "${pilot?.name ?: "Nobody"}: $status"
}

fun main() {
    val banner = """
        |Mission café 🚀
        |Signal: λ 𝌆
    """.trimMargin()
    val pilots = listOf(Pilot("Ada"), Pilot("Lin", active = false))
    pilots.filter { it.active }.forEach { pilot ->
        println("$banner\n${describe(pilot)}")
    }
}
